### DISCUSSION TITLE
Restart Machine instance if the MachineDeployment spec changed

### DISCUSSION THREAD

**Discussion by mirza-ali-ctct** [kubernetes-sigs/cluster-api#8821] [General]
We are adding custom labels in `MachineDeployment` under `spec.template.metadata.label`, the new labels are not propagated to `Machine` instances. But if we manually restart the machines of `MachineSet`, it works fine.  Can we force restart the machines if the labels?

**Comment by killianmuldoon** [comment]
>We are adding custom labels in MachineDeployment under spec.template.metadata.label, the new labels are not propagated to Machine instances. But if we manually restart the machines of MachineSet, it works fine. Can we force restart the machines if the labels?

It would be good to move this to an issue with more details such as: 
- What version of cluster API are you using?
- What infrastructure provider are you using?
- What labels are you trying to propagate?
- What do you mean by "manually restart"?
